fix(ivy): don't project removed placeholders with runtime i18n

ocombe opened the pull request:
When translated content was projected, all of the content was reappended, even the placeholders that had been removed in the translation.
To avoid that we added a new flag on `TNode` that specifies that a node is detached, in which case it should be ignored by the projection.

Ignore the first 3 commits that belong to the PRs #30696 and #30782. It should only be merged after these two.

mhevery commented on packages/core/test/acceptance/i18n_spec.ts:
```diff
@@ -902,8 +902,7 @@ onlyInIvy('Ivy i18n logic').describe('runtime i18n', () => {
 
-    // FW-1319 Runtime i18n should be able to remove projected placeholders
-    xit('should be able to remove projected placeholders', () => {
+    it('should be able to remove projected placeholders', () => {
```
You changed the behavior but no new test is added. Could we add a test demonstrating what code paths did not work before?

ocombe replied:
I enabled the test that was failing and was disabled previously, this test will use the code paths that did not work before

mhevery replied:
Ohh! I missed the `x` part I thought you only change the text.

ocombe committed:
fix(ivy): don't project removed placeholders with runtime i18n

When translated content was projected, all of the content was reappended, even the placeholders that had been removed in the translation.
To avoid that we added a new flag on `TNode` that specifies that a node is detached, in which case it should be ignored by the projection.

FW-1319 #resolve

alxhub reviewed (approved):
👍 for compiler (no compiler changes anymore)
